### ISSUE TITLE
Leveraging Bagging with HoeffdingTree

### ISSUE THREAD

**Issue by jmread** [scikit-multiflow/scikit-multiflow#21]
`LeveragingBagging` not working with `HoeffdingTree`, see for example `_test_prequential_bagging.py`. No error message given:

```
root - INFO - Prequential Evaluation
root - INFO - Generating 2 targets.
root - INFO - Pre-training on 2000 samples.
root - INFO - Evaluating...

root - INFO - Evaluation time: 0.000 s
root - INFO - Total instances: 2000
root - INFO - Global performance:
root - INFO - Learner 0 - Accuracy     : 0.000
root - INFO - Learner 0 - Kappa        : 0.000
root - INFO - Learner 0 - Kappa T      : 0.000
```
(But it works with `KNN` no problem).

**Comment by jacobmontiel** [comment]
The root cause of this issue seems to be:

```python
probs = self.predict_proba(X)
```

inside of the `predict(X)` method of `LeverageBagging`. This should be fixed by #26. It is strange that the `NoImplementationError` in `HoeffdingTree.predict_proba()` was not raised.

A secondary issue was found and fixed related to an error when getting the stream attribute `target_values`.

The fix is currently available in the `stream_genrators` branch.

**Comment by jmread** [comment]
Indeed, it seems to work now .. some of the time (I am using `_test_prequential_bagging.py` to test as earlier):

```
root - INFO - Prequential Evaluation
root - INFO - Evaluating 1 target(s).
root - INFO - Pre-training on 2000 samples.
root - INFO - Evaluating...
root - INFO - 10.0%
'NoneType' object is not iterable
root - INFO - Evaluation time: 8.396 s
root - INFO - Total samples: 2074
root - INFO - Global performance:
root - INFO - Model 0 - Accuracy     : 0.757
root - INFO - Model 0 - Kappa        : 0.634
```

Other times, I get
```
    self.n_detected_changes += 1
TypeError: unsupported operand type(s) for +=: 'NoneType' and 'int'
```

And on the other hand it only works under `WaveformGenerator` and not `SEAGenerator` but that *may* be a different issue.
